installer/src/uz801.rs

```diff
@@ -0,0 +1,234 @@
+            .body(r#"{"funcNo":2001}"#)
+            .send()
+            .await;
+        // Ignore any errors - the device will reboot and connection will be lost
+    });
+
+    Ok(())
+}
+
+async fn wait_for_adb() -> Result<ADBUSBDevice> {
+    const MAX_ATTEMPTS: u32 = 30; // 30 seconds
+    let mut attempts = 0;
+
+    // Wait a bit for the reboot to start
+    sleep(Duration::from_secs(10)).await;
+
+    loop {
+        if attempts >= MAX_ATTEMPTS {
+            anyhow::bail!("Timeout waiting for ADB connection after USB debug activation");
+        }
+
+        // UZ801 USB vendor and product IDs.
+        // TODO: Research if other variants use different IDs.
+        match ADBUSBDevice::new(0x05c6, 0x90b6) {
+            Ok(mut device) => {
+                // Test ADB connection
+                if test_adb_connection(&mut device).await.is_ok() {
+                    return Ok(device);
+                }
+            }
+            Err(RustADBError::DeviceNotFound(_)) => {
+                // Device not ready yet, continue waiting
+            }
+            Err(e) => {
+                anyhow::bail!("ADB connection error: {}", e);
+            }
+        }
+
+        sleep(Duration::from_secs(1)).await;
+        attempts += 1;
+    }
+}
+
+async fn test_adb_connection(adb_device: &mut ADBUSBDevice) -> Result<()> {
+    let mut buf = Vec::<u8>::new();
+    adb_device.shell_command(&["echo", "test"], &mut buf)?;
+    let output = String::from_utf8_lossy(&buf);
+    if output.contains("test") {
+        Ok(())
+    } else {
+        anyhow::bail!("ADB connection test failed")
+    }
+}
+
+async fn install_rayhunter_files(adb_device: &mut ADBUSBDevice) -> Result<()> {
+    // Create rayhunter directory
+    let mut buf = Vec::<u8>::new();
+    adb_device.shell_command(&["mkdir", "-p", "/data/rayhunter"], &mut buf)?;
+
+    // Remount system as writable
+    adb_device.shell_command(&["mount", "-o", "remount,rw", "/system"], &mut buf)?;
+
+    // Install rayhunter daemon binary with verification
+    let rayhunter_daemon_bin = include_bytes!(env!("FILE_RAYHUNTER_DAEMON"));
+    install_file(
+        adb_device,
+        "/data/rayhunter/rayhunter-daemon",
+        rayhunter_daemon_bin,
+    )?;
+
+    // Install config file
+    let config_content = crate::CONFIG_TOML.replace("#device = \"orbic\"", "device = \"uz801\"");
+    let mut config_data = config_content.as_bytes();
+    adb_device.push(&mut config_data, &"/data/rayhunter/config.toml")?;
+
+    // Make daemon executable
+    let mut buf = Vec::<u8>::new();
+    adb_device.shell_command(
+        &["chmod", "755", "/data/rayhunter/rayhunter-daemon"],
+        &mut buf,
+    )?;
+
+    Ok(())
+}
+
+/// Transfer a file to the device's filesystem with adb push.
+/// Validates the file sends successfully to /data/local/tmp
+/// before overwriting the destination.
+fn install_file(adb_device: &mut ADBUSBDevice, dest: &str, payload: &[u8]) -> Result<()> {
```

Comment:
is there any way to deduplicate this code with pinephone.rs?

Reply:
Hm, I can think of a way to dedupe it, but I didn't want to make it too ambitious before I could prove it works.